bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/PerChannelBookieClient.java

```diff
@@ -832,28 +836,7 @@ private void writeAndFlush(final Channel channel,
         }
 
         try{
-            channel.writeAndFlush(request)
-                .addListener(new ChannelFutureListener() {
-                        @Override
-                        public void operationComplete(ChannelFuture future)
-                                throws Exception {
-                            if (future.isSuccess()) {
-                                if (LOG.isDebugEnabled()) {
-                                    LOG.debug("Successfully wrote request {} to {}",
-                                              requestToString(request),
-                                              channel.remoteAddress());
-                                }
-                            } else {
-                                if (!(future.cause()
-                                      instanceof ClosedChannelException)) {
-                                    LOG.warn("Writing request {} to {} failed : ",
-                                             requestToString(request),
-                                             channel, future.cause());
-                                }
-                                errorOut(key);
```

Comment:
Are we dropping this errorOut in case of write error?

Reply:
@eolivelli Yes. Any write error will result in an exception on the channel, which will result in the channel being closed, which will result in all active keys being errored out.

Reply:
@ivankelly do we have some sort of test case about this case ?
Anyway your explanation is very clear to me